pkg/apis/pipeline/v1/param_types.go

```diff
@@ -83,7 +83,7 @@ func (pp *ParamSpec) SetDefaults(context.Context) {
 	case pp.Type != "":
 		// If param type is provided by the author, do nothing but just set default type for PropertySpec in case `properties` section is provided.
 		pp.setDefaultsForProperties()
-	case pp.Properties != nil:
```

Comment:
Thank you @JeromeJu . Could you please help me understand why do we want to prevent the empty `map[string]*v1.PropertySpec{}`  from being cast to ParamTypeObject? It's kind of weird that we set a map in the Propoerties then the type turns out to be "string". (Did I miss some context information?)

Reply:
+1, it'd be surprising to have an empty property spec but then the type is string -- wondering if it makes sense to fix the conversion that you mentioned in https://github.com/tektoncd/pipeline/pull/6570#issuecomment-1518036048?

Reply:
Thanks both. I wonder it would make sense for https://google.github.io/styleguide/go/decisions#nil-slices, it suggests not to differentiate empty map and nil, which points to that we should not further set empty maps back to nil?

Reply:
Nice doc @JeromeJu !

The doc refers to nil slices instead of nil map (which is our case) If I'm not mistaken?

In the meantime, I assume there are functional differences between `nil` and `map[string]*v1.PropertySpec{}`? Because the map struct itself implies it is an object parameter, unless I missed something.

Reply:
as noted in the other PR, please add this context to the commit message and PR description 

also wondering if the same argument can be made for empty slices and nil in array params/results 🤔

Reply:
Thanks both @jerop and @XinruZhang .

I think the difference between empty map & array makes sense. TY @XinruZhang  for the catch. While according to the doc, the empty slices should be treated as the same for nil, but I am not sure in this case if we should treat them as the same in this case cc @lbernick ? 
I wonder if `if taskResult.Properties != nil && len(taskResult.Properties) !=0` should be the condition in this block?